### PR TITLE
Groups

### PR DESCRIPTION
**Pull request by AlexChalk**
Users can create and join groups

### CODE REVIEW

**Comment by enonnai** [comment]
Guys, I created a group with useraccount1 and joined, then I logged in with useraccount2 and clicked on the group name(step1 screenshot), and the following error occurs as it points to a wrong path groups/groups/1 (step2 screenshot):
step_1
step_2

**Comment by AlexChalk** [comment]
Error occurs when user navigates to `groups/` instead of `groups`.

**Comment by AlexChalk** [comment]
@enonnai: Bug has been fixed!

**Comment by enonnai** [comment]
Awesome inter-teams work guys!

**Comment by DSeanGray** [comment]
hey, guys! code looks good there are a lot of empty or unnecessary files and commented out code in here though. can you remove them please. then it's all good to go 👍

**Review by dvidgry — DISMISSED** [review]
please remove the commented out code and redundant files and it all god to go :) good work guys 👍

**Comment by AlexChalk** [comment]
@DSeanGray the stuff you're referring to is auto-generated so i'd rather leave it in.

**Review by anaalta — APPROVED** [review]
Looks good (with David's remark that there are some files with commented code only - the css, the js files)
:shipit:

**Review by enonnai — APPROVED** [review]
Good stuff :shipit:

File: app/controllers/groups_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def index
+    @groups = Group.all
+  end
+
+  def show
+    @group = Group.find(params[:id])
+    @users = @group.users
+  end
+
+  def update
```
Comment (dbugsy) [inline app/controllers/groups_controller.rb]: This is not updating a Group, it is creating a Membership. Also, it is non-idempotent, meaning you could call this action twice and you would get 2 similar Membership rows. PUT http requests should be idempotent. http://restcookbook.com/HTTP%20Methods/put-vs-post/

Not a blocker, but feels un-railsy - you could consider a refactor.

File: app/models/membership.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Membership < ApplicationRecord
+  belongs_to :user
+  belongs_to :group
+  validates_uniqueness_of :user_id, scope: :group_id
```
Comment (dbugsy) [inline app/models/membership.rb:4]: very railsy - great!

File: spec/web_helpers.rb [hunk 3]
```diff
@@ -23,6 +23,13 @@ def create_a_post(text)
   click_button "Submit"
 end
 
+def create_a_group(title)
```
Comment (dbugsy) [inline spec/web_helpers.rb]: nice use of argument

**Review by dbugsy — APPROVED** [review]
Happy to let this through, but you could definitely improve this as follows:

`#update` Group is a little strange - looks like you could use the Membership `#create` action instead?

Commented code is a problem, there are a bunch of generated files here that are unused. It's good practice to delete those.

**Pull request merged**